ambry-store/src/main/java/com.github.ambry.store/IndexSegment.java

```diff
@@ -311,8 +315,8 @@ private int numberOfEntries(ByteBuffer mmap) {
 
   private StoreKey getKeyAt(ByteBuffer mmap, int index) throws IOException {
     mmap.position(
-        Version_Field_Length + Key_Size_Field_Length + Value_Size_Field_Length + Log_End_Offset_Field_Length + (index
-            * (keySize + valueSize)));
+        Version_Field_Length + Key_Size_Field_Length + Value_Size_Field_Length + Log_End_Offset_Field_Length + +
```

Comment:
Does this need to be index-segment version specific? E.g., `if (ver1) {} else if (ver2) {}`